django/contrib/postgres/forms/hstore.py

```diff
@@ -23,6 +23,8 @@ def prepare_value(self, value):
     def to_python(self, value):
         if not value:
             return {}
+        if isinstance(value, dict):
+            value = json.dumps(value)
         try:
             value = json.loads(value)
         except ValueError:
```

Comment:
This might be better:
```python
if not isinstance(value, dict):
    try:
        value = json.loads(value)
    except ValueError:
        raise ValidationError(
            self.error_messages['invalid_json'],
            code='invalid_json',
        )
# Cast everything to strings for ease.
for key, val in value.items():
    value[key] = six.text_type(val)
```